fault-proof/src/proposer.rs

```diff
@@ -480,53 +491,401 @@ where
+            }
+            TaskInfo::GameResolution => {
+                ProposerGauge::GameResolutionError.increment(1.0);
+            }
+            TaskInfo::BondClaim => {
+                ProposerGauge::BondClaimingError.increment(1.0);
+            }
+        }
+        Ok(())
+    }
+
+    /// Spawn pending operations if not already running
+    async fn spawn_pending_operations(&self) -> Result<()> {
+        // Check if we should create a game and spawn task if needed
+        if !self.has_active_task_of_type(&TaskInfo::GameCreation { block_number: U256::ZERO }).await
+        {
+            match self.spawn_game_creation_task().await {
+                Ok(true) => tracing::info!("Successfully spawned game creation task"),
+                Ok(false) => {
+                    tracing::debug!("No game creation needed - proposal interval not elapsed")
+                }
+                Err(e) => tracing::warn!("Failed to spawn game creation task: {:?}", e),
+            }
+        } else {
+            tracing::info!("Game creation task already active");
+        }
+
+        // Check if we should defend games
+        match self.spawn_game_defense_tasks().await {
+            Ok(true) => tracing::info!("Successfully spawned game defense task"),
+            Ok(false) => tracing::debug!("No games need defense or task already active"),
+            Err(e) => tracing::warn!("Failed to spawn game defense tasks: {:?}", e),
+        }
+
+        // Check if we should resolve games
+        if !self.has_active_task_of_type(&TaskInfo::GameResolution).await {
+            match self.spawn_game_resolution_task().await {
+                Ok(true) => tracing::info!("Successfully spawned game resolution task"),
+                Ok(false) => tracing::debug!("No games need resolution"),
+                Err(e) => tracing::warn!("Failed to spawn game resolution task: {:?}", e),
+            }
+        } else {
+            tracing::info!("Game resolution task already active");
+        }
+
+        // Check if we should claim bonds
+        if !self.has_active_task_of_type(&TaskInfo::BondClaim).await {
+            match self.spawn_bond_claim_task().await {
+                Ok(true) => tracing::info!("Successfully spawned bond claim task"),
+                Ok(false) => tracing::debug!("No bonds available to claim"),
+                Err(e) => tracing::warn!("Failed to spawn bond claim task: {:?}", e),
+            }
+        } else {
+            tracing::info!("Bond claim task already active");
+        }
+
+        Ok(())
+    }
+
+    /// Check if there's an active task of the given type
+    async fn has_active_task_of_type(&self, task_type: &TaskInfo) -> bool {
+        let tasks = self.tasks.lock().await;
+        tasks
+            .values()
+            .any(|(_, info)| std::mem::discriminant(info) == std::mem::discriminant(task_type))
+    }
+
+    /// Log current task statistics
+    async fn log_task_stats(&self) {
```

Comment:
in prod, the interval is pretty long, so i think we should also log the task types being worked on.